### ISSUE TITLE
"InstanceMetadataServiceConfiguration" as invalid property for AWS::SageMaker::NotebookInstance

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#2449]
### CloudFormation Lint Version

v0.69.1

### What operating system are you using?

Mac

### Describe the bug

"InstanceMetadataServiceConfiguration" as invalid property for AWS::SageMaker::NotebookInstance

### Expected behavior

This value, according to the docs, should exist and I shouldn't get an error.

### Reproduction template

```yaml
Description: "Create basic notebook instance"
Resources:
  BasicNotebookInstance:
    Type: "AWS::SageMaker::NotebookInstance"
    Properties:
      InstanceType: "ml.t2.large"
      RoleArn: !GetAtt ExecutionRole.Arn
      InstanceMetadataServiceConfiguration: {}
  ExecutionRole: 
    Type: "AWS::IAM::Role"
    Properties: 
      AssumeRolePolicyDocument: 
        Version: "2012-10-17"
        Statement: 
          - 
            Effect: "Allow"
            Principal: 
              Service: 
                - "sagemaker.amazonaws.com"
            Action: 
              - "sts:AssumeRole"
      Path: "/"
      ManagedPolicyArns:
        - !Sub "arn:${AWS::Partition}:iam::aws:policy/AmazonSageMakerFullAccess"
Outputs:
  BasicNotebookInstanceId:
    Value: !Ref BasicNotebookInstance
```

**Comment by kddejong** [comment]
Replaces issue https://github.com/aws-cloudformation/cfn-lint-visual-studio-code/issues/239

**Comment by kddejong** [comment]
This has been resolved and the property is now in the spec
